src/main/java/com/minecolonies/api/util/constant/IToolType.java

```diff
@@ -26,5 +29,27 @@ public interface IToolType
      * @return The text displayed to the user.
      */
     Component getDisplayName();
+
+    /**
+     * Whether the item stack works as the tool.
+     *
+     * @param itemStack to test
+     * @return Whether the item stack works as the tool.
+     */
+    Boolean checkIsTool(ItemStack itemStack);
+
+    /**
+     * The level the item stack is.
+     *
+     * @param itemStack to test
+     * @return The level the item stack is.
+     */
+    int getMiningLevel(ItemStack itemStack);
+
+    class IToolTypeComparator implements Comparator<IToolType> {
```

Comment:
Javadoc and formatting

Reply:
Done. Though the CheckStyle config fails to load for me.